### PR TITLE
fix(build): add elasticsearch feature to block serialize

### LINKED ISSUE #6695: Build failure when elasticsearch feature is enabled
Reported by John in slack. When i execute this command to install zebra:

```
cargo install --features="elasticsearch" --locked --git https://github.com/ZcashFoundation/zebra --tag v1.0.0-rc.8 zebrad
```

I get the following build error:
```
...
   Compiling zebra-state v1.0.0-beta.24 (/home/oxarbitrage/.cargo/git/checkouts/zebra-d0d93d7ded255019/0190882/zebra-state)
error[E0277]: the trait bound `zebra_chain::block::Block: Serialize` is not satisfied
   --> zebra-state/src/service/finalized_state.rs:388:23
    |
388 |                 .push(serde_json::json!(block).to_string());
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^
    |                       |
    |                       the trait `Serialize` is not implemented for `zebra_chain::block::Block`
    |                       required by a bound introduced by this call
    |
    = help: the following other types implement trait `Serialize`:
              &'a T
              &'a mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 276 others
    = note: required for `Arc<zebra_chain::block::Block>` to implement `Serialize`
    = note: 2 redundant requirements hidden
    = note: required for `&&Arc<zebra_chain::block::Block>` to implement `Serialize`
note: required by a bound in `to_value`
   --> /home/oxarbitrage/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.96/src/value/mod.rs:942:8
    |
942 |     T: Serialize,
    |        ^^^^^^^^^ required by this bound in `to_value`
    = note: this error originates in the macro `json_internal` which comes from the expansion of the macro `serde_json::json` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `zebra-state` due to previous error
error: failed to compile `zebrad v1.0.0-rc.8 (https://github.com/ZcashFoundation/zebra?tag=v1.0.0-rc.8#01908822)`, intermediate artifacts can be found at `/tmp/cargo-install27LFAQ`
$
```

This is also happening without `--locked` argument.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

Close https://github.com/ZcashFoundation/zebra/issues/6695

## Solution

Add elasticsearch feature to block serialization.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

**Review by teor2345 — COMMENTED** [review]
Did you manually test the same command that failed in the ticket?

**Comment by oxarbitrage** [comment]
This is working:

```
 cargo install zebrad  --locked --git https://github.com/ZcashFoundation/zebra --branch es-serialize-buld-error zebrad
```

**Comment by teor2345** [comment]
Are you missing `--features="elasticsearch"`?

**Comment by oxarbitrage** [comment]
Yea, sorry, i just noticed that. Trying again.

**Comment by teor2345** [comment]
> Are you missing `--features="elasticsearch"`?

If this was a production feature, we'd add a CI test that specifically tried this compile. But since it's experimental it's ok to break it in some circumstances.

**Comment by oxarbitrage** [comment]
I changed it back to draft as it seems the changes did not worked with the full correct command i wanted to fix:

```
$ cargo install zebrad --features="elasticsearch" --locked --git https://github.com/ZcashFoundation/zebra --branch es-serialize-buld-error zebrad
...
error[E0277]: the trait bound `zebra_chain::block::Block: Serialize` is not satisfied
   --> zebra-state/src/service/finalized_state.rs:388:23
    |
388 |                 .push(serde_json::json!(block).to_string());
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^
    |                       |
    |                       the trait `Serialize` is not implemented for `zebra_chain::block::Block`
    |                       required by a bound introduced by this call
    |
    = help: the following other types implement trait `Serialize`:
              &'a T
              &'a mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 276 others
...
```

File: zebra-chain/src/block.rs [hunk 1]
```diff
@@ -43,7 +43,10 @@ pub use arbitrary::LedgerState;
 
 /// A Zcash block, containing a header and a list of transactions.
 #[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
```
Comment (teor2345) [inline zebra-chain/src/block.rs:47]: Every crate that uses the `elasticsearch` feature needs to have it declared in its `Cargo.toml`, and it needs to activate that feature in the crates that it depends on. (If they have that feature.)

You can use `getblocktemplate-rpcs` as an example:
https://github.com/ZcashFoundation/zebra/blob/2e379811c3a8f4dd74c8e0e735738dcbe52d4288/zebrad/Cargo.toml#L28-L35

The list of activated dependencies or features can be empty if there's nothing extra needed.
https://github.com/ZcashFoundation/zebra/blob/2e379811c3a8f4dd74c8e0e735738dcbe52d4288/zebra-chain/Cargo.toml#L20-L23

**Review by teor2345 — COMMENTED** [review]
We'll need to chain the feature activations between different crates.

File: zebra-chain/src/block.rs (hunk 1 above)
Comment (oxarbitrage) [reply zebra-chain/src/block.rs:47]: Yea, thats needed. And some other changes. They were done [here](https://github.com/ZcashFoundation/zebra/pull/6709/commits/a809644bd9ac088af681412143a0d560352243de)

This command is now worrking:

```
cargo install zebrad --features="elasticsearch" --git https://github.com/ZcashFoundation/zebra --branch es-serialize-buld-error zebrad
```

I am moving this PR out of draft.

Comment (teor2345) [reply zebra-chain/src/block.rs:47]: We're currently in a release freeze, so I'm going to mark this PR to merge after the release.

File: zebra-state/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ proptest-impl = [
 elasticsearch = [
     "dep:elasticsearch",
     "dep:serde_json",
+    "zebra-chain/elasticsearch",
```
Comment (teor2345) [inline zebra-state/Cargo.toml:33]: Optional: we could make the same change in `zebrad`

The `cargo` feature resolver should work it out automatically, but it's had bugs in the past.

https://github.com/ZcashFoundation/zebra/blob/2e379811c3a8f4dd74c8e0e735738dcbe52d4288/zebrad/Cargo.toml#L37-L39

**Review by teor2345 — DISMISSED** [review]
Looks good, thanks!

I'm assuming you did the manual test again?

**Review by teor2345 — COMMENTED** [review]
Thanks!

I want to approve, but someone needs to manually test that this works first.

**Comment by oxarbitrage** [comment]
I tested this manually and it is working.

**Review by teor2345 — APPROVED** [review]

**Pull request merged**